### PR TITLE
[Dns] Resolve Apiview Comments

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by dvbb** [comment]
Due to TestFramework still occurring Castle.DynamicProxy.Generators.GeneratorException, I had to test these cases elsewhere without TestFramework.
`DnsZone` and all `FooRecordSet`(Aaaa,A,Caa,Cname,MX,NS,Ptr,Soa,Srv,Txt) test cases passed.

**Review by ArthurMa1978 — APPROVED** [review]

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/MXRecordSetData.cs [hunk 1]
```diff
@@ -35,19 +36,19 @@ public MxRecordSetData()
...
-            MxRecords = mxRecords;
+            MXRecords = mxRecords;
         }
 
         /// <summary> The etag of the record set. </summary>
-        public string Etag { get; set; }
+        public ETag? ETag { get; set; }
         /// <summary> The metadata attached to the record set. </summary>
         public IDictionary<string, string> Metadata { get; }
         /// <summary> The TTL (time-to-live) of the records in the record set. </summary>
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/MXRecordSetData.cs:54]: Ttl

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 2]
```diff
@@ -300,11 +300,11 @@ public partial class RecordSetData : Azure.ResourceManager.Models.ResourceData
         public System.Collections.Generic.IList<Azure.ResourceManager.Dns.Models.ARecord> ARecords { get { throw null; } }
         public System.Collections.Generic.IList<Azure.ResourceManager.Dns.Models.CaaRecord> CaaRecords { get { throw null; } }
         public string Cname { get { throw null; } set { } }
-        public string ETag { get { throw null; } set { } }
+        public Azure.ETag? ETag { get { throw null; } set { } }
         public string Fqdn { get { throw null; } }
         public System.Collections.Generic.IDictionary<string, string> Metadata { get { throw null; } }
-        public System.Collections.Generic.IList<Azure.ResourceManager.Dns.Models.MxRecord> MxRecords { get { throw null; } }
-        public System.Collections.Generic.IList<Azure.ResourceManager.Dns.Models.NsRecord> NsRecords { get { throw null; } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.Dns.Models.MXRecord> MxRecords { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:306]: the property name should be `MXRecords`

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 3]
```diff
@@ -300,11 +300,11 @@ public partial class RecordSetData : Azure.ResourceManager.Models.ResourceData
...
+        public System.Collections.Generic.IList<Azure.ResourceManager.Dns.Models.NSRecord> NsRecords { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:307]: the property name should be `NSRecords`

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/RecordSetMXResource.cs [hunk 4]
```diff
@@ -145,11 +145,11 @@ public virtual Response<RecordSetMxResource> Get(CancellationToken cancellationT
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         public virtual async Task<ArmOperation> DeleteAsync(WaitUntil waitUntil, string ifMatch = null, CancellationToken cancellationToken = default)
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/RecordSetMXResource.cs:146]: all `ifMatch` should be `ETag` type

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 5]
```diff
@@ -561,21 +561,21 @@ public partial class DnsZonePatch
...
+        public NSRecord() { }
+        public string NsdName { get { throw null; } set { } }
     }
     public partial class PtrRecord
     {
         public PtrRecord() { }
-        public string Ptrdname { get { throw null; } set { } }
+        public string PtrdName { get { throw null; } set { } }
     }
     public partial class SoaRecord
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:580]: Prepend `Dns` 

Also for other models with simple name, like the `ZoneType`

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs (hunk 3 above)
Comment (dvbb) [reply sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs]: If its list is called `MXRecords`, then it should make sense for its single model called `MXRecord`

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs (hunk 5 above)
Comment (dvbb) [reply sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:585]: NsdName => DnsNsdName
Ptrdname => DnsPtrdname
ZoneType => DnsZoneType
ZoneListResult => DnsZoneListResult (Considering that all other `Zone` prefixed models are named `DnsZone`)

File: sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsARecord.cs [hunk 6]
```diff
@@ -8,16 +8,16 @@
...
-        /// <summary> Initializes a new instance of ARecord. </summary>
-        public ARecord()
+        /// <summary> Initializes a new instance of DnsARecord. </summary>
+        public DnsARecord()
         {
         }
 
-        /// <summary> Initializes a new instance of ARecord. </summary>
+        /// <summary> Initializes a new instance of DnsARecord. </summary>
         /// <param name="ipv4Address"> The IPv4 address of this A record. </param>
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsARecord.cs:19]: Can this be `IPAddress` type

File: sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsAaaaRecord.cs [hunk 7]
```diff
@@ -8,16 +8,16 @@
...
-        /// <summary> Initializes a new instance of AaaaRecord. </summary>
-        public AaaaRecord()
+        /// <summary> Initializes a new instance of DnsAaaaRecord. </summary>
+        public DnsAaaaRecord()
         {
         }
 
-        /// <summary> Initializes a new instance of AaaaRecord. </summary>
+        /// <summary> Initializes a new instance of DnsAaaaRecord. </summary>
         /// <param name="ipv6Address"> The IPv6 address of this AAAA record. </param>
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsAaaaRecord.cs:19]: Can this be `IPAddress` type

File: sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsSoaRecord.cs [hunk 8]
```diff
@@ -8,22 +8,22 @@
...
         {
         }
 
-        /// <summary> Initializes a new instance of SoaRecord. </summary>
+        /// <summary> Initializes a new instance of DnsSoaRecord. </summary>
         /// <param name="host"> The domain name of the authoritative name server for this SOA record. </param>
         /// <param name="email"> The email contact for this SOA record. </param>
         /// <param name="serialNumber"> The serial number for this SOA record. </param>
         /// <param name="refreshTime"> The refresh value for this SOA record. </param>
         /// <param name="retryTime"> The retry time for this SOA record. </param>
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsSoaRecord.cs:23]: What's the unit?

File: sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsSoaRecord.cs [hunk 9]
```diff
@@ -8,22 +8,22 @@
...
         /// <param name="expireTime"> The expire time for this SOA record. </param>
```
Comment (ArthurMa1978) [inline sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsSoaRecord.cs:24]: What's the unit?

File: sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsSoaRecord.cs [hunk 10]
```diff
@@ -8,22 +8,22 @@
...
+        public DnsSoaRecord()
         {
         }
 
-        /// <summary> Initializes a new instance of SoaRecord. </summary>
+        /// <summary> Initializes a new instance of DnsSoaRecord. </summary>
         /// <param name="host"> The domain name of the authoritative name server for this SOA record. </param>
         /// <param name="email"> The email contact for this SOA record. </param>
         /// <param name="serialNumber"> The serial number for this SOA record. </param>
         /// <param name="refreshTime"> The refresh value for this SOA record. </param>
```
[the remark this answers is not in the document]
Comment (dvbb) [reply sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsSoaRecord.cs:22]: > Prepend Dns. Also for other models with simple name, like the ZoneType

According to the previous comments, add the `Dns` prefix to `PtrRecord`. Ptr is one of ten RecordSets (Aaaa,A,Caa,Cname,MX,NS,Ptr,Soa,Srv,Txt), so I prefixed the rest of the RecordSets with Dns.

**Pull request merged**